### PR TITLE
Remove EventBus

### PR DESCRIPTION
**Pull request by amanmahajan7**
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often

This is only recommended if there is no other solution and I think it is one of those situations where other patterns do not work. Tested and there was no performance impact

### CODE REVIEW

File: src/hooks/useLatest.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { useRef, useEffect } from 'react';
+
+export function useLatest<T>(value: T) {
```
Comment (nstepien) [inline src/hooks/useLatest.ts:3]:
```suggestion
// https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
export function useLatest<T>(value: T) {
```

File: src/hooks/useLatest.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  const ref = useRef<T>(value);
+  useEffect(() => {
```
Comment (nstepien) [inline src/hooks/useLatest.ts:5]: I wonder if using `useLayoutEffect` would a bit safer as it's the "sync" version of `useEffect`.

File: src/hooks/useLatest.ts (hunk 1 above)
Comment (nstepien) [inline src/hooks/useLatest.ts:3]: Do we only use it for functions?
Maybe we could add a `useCallback` so we can skip the `current` part.

File: src/GroupRow.tsx [hunk 3]
```diff
@@ -1,9 +1,26 @@
-import React, { memo } from 'react';
+import React, { memo, MutableRefObject } from 'react';
 import clsx from 'clsx';
-import { GroupRowRendererProps } from './types';
+
+import { CalculatedColumn, Position, SelectRowEvent } from './types';
```
Comment (nstepien) [inline src/GroupRow.tsx:4]:
```suggestion
import { CalculatedColumn, Position, SelectRowEvent, Omit } from './types';
```

**Review by nstepien — CHANGES_REQUESTED** [review]

File: src/hooks/useLatest.ts (hunk 2 above)
Comment (amanmahajan7) [reply src/hooks/useLatest.ts]: I am not 100% sure. React docs uses `useEffect` in the example. Also these function are only called when user performs an action (apart from `selectCell` which can be called using gridRef). I think we can use `useEffect` for now and revisit if we find bugs.

File: src/Cell.tsx [hunk 4]
```diff
@@ -39,7 +40,7 @@ function Cell<R, SR>({
   );
 
   function selectCell(openEditor?: boolean) {
```
Comment (nstepien) [inline src/Cell.tsx:42]: Isn't this the `wrapper` instead? 🤔

File: src/hooks/useLatestFunc.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { useRef, useEffect, useCallback } from 'react';
+
+// https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function useLatestFunc<T extends (...args: any[]) => any>(fn: T) {
+  const ref = useRef<T>(fn);
```
Comment (nstepien) [inline src/hooks/useLatestFunc.ts:6]:
```suggestion
  const ref = useRef(fn);

```

File: src/types.ts [hunk 6]
```diff
@@ -145,9 +144,10 @@ export interface CellRendererProps<TRow, TSummaryRow = unknown> extends Omit<Rea
   isDraggedOver: boolean;
   isCellSelected: boolean;
   isRowSelected: boolean;
-  eventBus: EventBus;
   dragHandleProps?: Pick<React.HTMLAttributes<HTMLDivElement>, 'onMouseDown' | 'onDoubleClick'>;
   onRowClick?: (rowIdx: number, row: TRow, column: CalculatedColumn<TRow, TSummaryRow>) => void;
+  selectCell: (position: Position, enableEditor?: boolean) => void;
+  selectRow: ({ rowIdx, checked, isShiftClick }: SelectRowEvent) => void;
```
Comment (nstepien) [inline src/types.ts:150]:
```suggestion
  selectRow: (selectRowEvent: SelectRowEvent) => void;
```

File: src/types.ts [hunk 7]
```diff
@@ -158,26 +158,13 @@ export interface RowRendererProps<TRow, TSummaryRow = unknown> extends Omit<Reac
...
-  childRows: readonly TRow[];
-  rowIdx: number;
-  top: number;
-  level: number;
-  selectedCellIdx?: number;
-  isExpanded: boolean;
-  isRowSelected: boolean;
-  eventBus: EventBus;
+  selectCell: (position: Position, enableEditor?: boolean) => void;
+  selectRow: ({ rowIdx, checked, isShiftClick }: SelectRowEvent) => void;
```
Comment (nstepien) [inline src/types.ts:167]:
```suggestion
  selectRow: (selectRowEvent: SelectRowEvent) => void;
```

**Review by nstepien — COMMENTED** [review]
Just a few nitpicks, wdyt?

**Review by nstepien — APPROVED** [review]

File: src/Cell.tsx (hunk 4 above)
Comment (amanmahajan7) [reply src/Cell.tsx]: Changed

**Review by nstepien — APPROVED** [review]

**Pull request merged**